### ISSUE TITLE
Decide if/how to resolve "content outside of landmarks" violations

### ISSUE THREAD

**Issue by domoscargin** [alphagov/govuk-frontend#6460]
## What
Some components are neither in the `<header>` or in `<main>` - things like the [phase banner and back links](https://govuk-frontend-review.herokuapp.com/full-page-examples/upload-your-photo-success). These [used to be flagged by axe as WCAG fails](https://github.com/alphagov/govuk-frontend/issues/1604) but have since been downgraded to "best practice".

We'd like to know if this is an issue for screen reader users and how important it is to fix.

Potentially, we could move the phase banner into the `<header>`, but we don't have as neat a plan for the Back link.

## Why
Using landmarks makes it easier for assistive technology to understand the structure of a page and navigate to useful sections of the page.

## Who needs to work on this
Accessibility specialist, developer

## Who needs to review this
Developer

## Done when

- [x] researched how important it is to screen reader users when some elements are not in landmarks and if the community ever came across issues with this in user research
- [x] decided what to do with the phase banner
- [x] decided what to do with the back link
- [x] documented all the possible options for the back link to help make a decision
- [x] thought about how to make accessibility flags like this more findable in our guidance
- [x] created GitHub issues for next actions

**Comment by selfthinker** [comment]
When the accessibility specialists met last week we talked about how much of an accessibility issue we think it is when something is outside of any landmarks. We more specifically talked about the phase banner and backlinks.

* We definitely don't think it is a WCAG fail. (Axe used to say that but it changed its classification to "best practice" at some point.)
* We are unsure how much of an accessibility issue it is as we don't know enough about how screen reader users might notice that or miss something because of it.
* We thought the phase banner could fit well into the header element.
* We had no idea what to do with the back link. It neither fits into main nor into the header. It might fit into its own nav element but that would be overkill. (It could theoretically fit well together with the breadcrumbs as a "structural navigation", but those are never together on the same page.)

I will create a document collecting all the potential options for the back links with their pros and cons so we can make a decision.

I have also read through all the GitHub issues related to this and couldn't find a single mention of an actual user having any issues because of this. All the feedback was there because of the Axe issue.
I might reach out on the x-gov Slack to ask if anyone came across an issues during user research.

**Comment by selfthinker** [comment]
I created an (internal) document listing [options for where to move the back link](https://docs.google.com/document/d/14rCitEzFMXautwLjtmKLNUt8DR7VJ-CeA-3E_MqncUw/edit?usp=sharing) with pros and cons for each and shared it with the team and other accessibility specialists.

Next we'll meet to discuss those options.

I also asked on the [cross-government Slack](https://ukgovernmentdigital.slack.com/archives/C6DMEH5R6/p1764155422217259) if anyone came across issues with real users. There haven't been any responses so far.

**Comment by selfthinker** [comment]
I've done a bit more background research throughout last week. I haven't learned anything new but what I knew was confirmed. A few points:

* Theory of landmarks and practice of landmarks are not fully consistent with each other. As always, what the spec says and how vendors implement it and how users use it can be different.
* If screen readers care about landmarks, they mostly care about the big ones: main (main), banner (header), contentinfo (footer) and navigation (nav), and out of those mostly about main. (An example of evidence of that is in this [discussion thread](https://webaim.org/discussion/mail_thread?thread=10090).)
* Screen reader users still mostly use headings to navigate through a page. From the [WebAIM screen reader survey](https://webaim.org/projects/screenreadersurvey10/#landmarks): "While 31.7% of respondents indicate that they always or often use landmarks when they are present, only 3.7% use this as a primary method for finding information on a lengthy web page."
* I couldn't find anything reliable that says where little snippets like the back link should go. The question is not just where it makes semantic sense, but much more where screen reader users expect to find it. There is no clear answer for that.

**Comment by selfthinker** [comment]
We met on Friday to make some decisions. (We have a decision log meeting later today, so these are potentially subject to change but that's not very likely.)

* We have decided that the phase banner should move into the header element. That makes the most sense as the phase banner is about the service as a whole.
* None of the options where to put the back link are great:
  * Moving it to the `header` is out of the question as it is more specific to the page you are on.
  * Wrapping it in an `aside` element is not great as it doesn't fit what the spec says.
  * Moving it to the `main` element would cause a WCAG fail of 2.4.1 Bypass Blocks.
  * Keeping it as it is (outside of any landmarks) will still cause an Axe best practice issue and therefore more support requests.
  * Wrapping it in a `nav` element means having a whole landmark for one single link which seems excessive. It is bad practice to have too many landmarks, although if a service has breadcrumbs on other pages this would at least not increase the general amount of landmarks per page (because breadcrumbs and back links are never together on the same page). It is possible that this might make it worse for screen reader users than not having this in a landmark.
* We have decided to keep the back link where it is especially because we haven't heard back from anyone that it is an issue with actual users. But we want to add some content onto the back link component page to a) let people know we are aware of the Axe issue and b) ask for feedback if people are aware of issues with actual users.
* If we ever want to move it after getting such feedback, the next best option would be to wrap it in a `nav` element as the back link is a navigational item.

I will create GitHub issues for those additional actions about adding guidance.

**Comment by selfthinker** [comment]
I have created an issue for [adding information to the back link component page](https://github.com/alphagov/govuk-design-system/issues/5024).

It is still unclear how the guidance for moving the phase banner into the header should be done.
At the very least it depends on the [guidance for changes to the page template](https://github.com/alphagov/govuk-design-system/issues/5037) as it should reference that and link back to it.
It should also be attached to the [content work for v6](https://github.com/alphagov/govuk-design-system/issues/4985).

**Comment by selfthinker** [comment]
> It is still unclear how the guidance for moving the phase banner into the header should be done. At the very least it depends on the [guidance for changes to the page template](https://github.com/alphagov/govuk-design-system/issues/5037) as it should reference that and link back to it. It should also be attached to the [content work for v6](https://github.com/alphagov/govuk-design-system/issues/4985).

I talked to Calvin yesterday and have now created a new issues for this (https://github.com/alphagov/govuk-design-system/issues/5123).
